### PR TITLE
find problem segfaults

### PR DESCRIPTION
**Pull request by swolferDF**
Find segfaults finishing programme with writer and reader 
In function updatewriterqos and updatereaderqos put args pyobject instead of Ref on c++ code

### CODE REVIEW

File: pyopendds/ext/_pyopendds.cpp [hunk 1]
```diff
@@ -432,53 +430,45 @@ bool update_writer_qos(Ref pyQos, DDS::DataWriterQos &qos)
     Ref pyhistorydepth;
 
 
-    // std::cerr << "get durability" << std::endl;
-    pydurability = PyObject_GetAttrString(*pyQos, "durability");
+    std::cerr << "get durability" << std::endl;
```
Comment (p-e-s) [inline pyopendds/ext/_pyopendds.cpp:433]: Can you remove this remaining log?

Comment (swolferDF) [reply pyopendds/ext/_pyopendds.cpp]: done

File: pyopendds/Publisher.py [hunk 2]
```diff
@@ -18,6 +18,7 @@ def __init__(self, participant: DomainParticipant, qos=None):
         create_publisher(self, participant)
 
     def create_datawriter(self, topic: Topic, qos=None) -> DataWriter:
+        print("[pyopendds] creat datawriter")
```
Comment (Sdpierret) [inline pyopendds/Publisher.py:21]: logs ?

File: pyopendds/Publisher.py [hunk 3]
```diff
@@ -18,6 +18,7 @@ def __init__(self, participant: DomainParticipant, qos=None):
...
         writer = DataWriter(self, topic, qos)
-        self.writers.append(writer)
+        # self.writers.append(writer)
```
Comment (Sdpierret) [inline pyopendds/Publisher.py:23]: dead code in comment

File: pyopendds/ext/_pyopendds.cpp [hunk 4]
```diff
@@ -240,16 +240,14 @@ PyObject* participant_cleanup(PyObject* self, PyObject* args)
         return nullptr;
     }
     pyparticipant++;
-
+
```
Comment (Sdpierret) [inline pyopendds/ext/_pyopendds.cpp:243]: trailing whitespaces

File: pyopendds/ext/_pyopendds.cpp [hunk 5]
```diff
@@ -240,16 +240,14 @@ PyObject* participant_cleanup(PyObject* self, PyObject* args)
...
     // Get DomainParticipant_var
     DDS::DomainParticipant* participant =
         get_capsule<DDS::DomainParticipant>(*pyparticipant);
-
     if (!participant) return nullptr;
 
     participant->delete_contained_entities();
-    participant_factory->delete_participant(participant);
-
+    // participant_factory->delete_participant(participant);
```
Comment (Sdpierret) [inline pyopendds/ext/_pyopendds.cpp:250]: dead code in comment

File: pyopendds/ext/_pyopendds.cpp [hunk 6]
```diff
@@ -420,9 +418,9 @@ void delete_datareader_var(PyObject* reader_capsule)
     }
 }
 
-bool update_writer_qos(Ref pyQos, DDS::DataWriterQos &qos)
-{
 
+ bool update_writer_qos(PyObject* pyQos, DDS::DataWriterQos &qos)
```
Comment (Sdpierret) [inline pyopendds/ext/_pyopendds.cpp:422]: space before "bool"

File: pyopendds/ext/_pyopendds.cpp [hunk 7]
```diff
@@ -666,7 +650,7 @@ PyObject* create_datawriter(PyObject* self, PyObject* args)
     if (set_capsule(*pydatawriter, datawriter, delete_datawriter_var)) {
         return nullptr;
     }
-
+
```
Comment (Sdpierret) [inline pyopendds/ext/_pyopendds.cpp:653]: trailing whitespace

File: pyopendds/ext/_pyopendds.cpp [hunk 8]
```diff
@@ -641,17 +625,17 @@ PyObject* create_datawriter(PyObject* self, PyObject* args)
     // Get Publisher
     DDS::Publisher* publisher = get_capsule<DDS::Publisher>(*pypublisher);
     if (!publisher) return nullptr;
-
+ 
     // Get Topic
     DDS::Topic* topic = get_capsule<DDS::Topic>(*pytopic);
     if (!topic) return nullptr;
-
+
```
Comment (Sdpierret) [inline pyopendds/ext/_pyopendds.cpp:632]: trailing whitespace

File: pyopendds/ext/_pyopendds.cpp [hunk 9]
```diff
@@ -641,17 +625,17 @@ PyObject* create_datawriter(PyObject* self, PyObject* args)
     // Get Publisher
     DDS::Publisher* publisher = get_capsule<DDS::Publisher>(*pypublisher);
     if (!publisher) return nullptr;
-
+
```
Comment (Sdpierret) [inline pyopendds/ext/_pyopendds.cpp:628]: trailing whitespace

**Review by Sdpierret — CHANGES_REQUESTED** [review]
Tu vas me haïr :/

**Review by p-e-s — APPROVED** [review]
Ok for me.

File: pyopendds/Publisher.py (hunk 3 above)
Comment (swolferDF) [reply pyopendds/Publisher.py]: done

File: pyopendds/ext/_pyopendds.cpp (hunk 5 above)
Comment (swolferDF) [reply pyopendds/ext/_pyopendds.cpp]: done

File: pyopendds/ext/_pyopendds.cpp (hunk 6 above)
Comment (swolferDF) [reply pyopendds/ext/_pyopendds.cpp]: done

**Review by Sdpierret — APPROVED** [review]

**Pull request merged**